source/isaaclab/isaaclab/sensors/frame_transformer/frame_transformer.py

```diff
@@ -416,45 +423,82 @@ def _update_buffers_impl(self, env_ids: Sequence[int]):
         self._data.target_pos_source[:] = target_pos_source.view(-1, total_num_frames, 3)
         self._data.target_quat_source[:] = target_quat_source.view(-1, total_num_frames, 4)
 
+    def _draw_lines(self, source_pos, target_pos):
+        # Calculate the direction vector and length
+        direction = target_pos - source_pos
+        length = torch.norm(direction, dim=-1)
+
+        # Calculate midpoint
+        midpoint = (source_pos + target_pos) / 2
+
+        # Get default direction (along z-axis)
+        default_direction = torch.tensor([0.0, 0.0, 1.0], device=self.device).expand(source_pos.size(0), -1)
+
+        # Normalize direction vector
+        direction_norm = normalize(direction)
+
+        # Calculate rotation from default direction to target direction
+        rotation_axis = torch.linalg.cross(default_direction, direction_norm)
+        rotation_axis_norm = torch.norm(rotation_axis, dim=-1)
+
+        # Handle case where vectors are parallel
+        mask = rotation_axis_norm > 1e-6
+        rotation_axis = torch.where(
+            mask.unsqueeze(-1),
+            normalize(rotation_axis),
+            torch.tensor([1.0, 0.0, 0.0], device=self.device).expand(source_pos.size(0), -1),
+        )
+
+        # Calculate rotation angle
+        cos_angle = torch.sum(default_direction * direction_norm, dim=-1)
+        cos_angle = torch.clamp(cos_angle, -1.0, 1.0)
+        angle = torch.acos(cos_angle)
+
+        # Convert to quaternion
+        quat = quat_from_angle_axis(angle, rotation_axis)
+
+        # Set scale (height of the cylinder)
+        scale = torch.ones((source_pos.size(0), 3), device=self.device)
+        scale[:, 2] = length  # Scale along z-axis
+
+        # Visualize the line
+        self._line_visualizer.visualize(translations=midpoint, orientations=quat, scales=scale)
+
     def _set_debug_vis_impl(self, debug_vis: bool):
         # set visibility of markers
         # note: parent only deals with callbacks. not their visibility
         if debug_vis:
             if not hasattr(self, "frame_visualizer"):
                 self.frame_visualizer = VisualizationMarkers(self.cfg.visualizer_cfg)
-
-                try:
-                    # isaacsim.util is not available in headless mode
-                    import isaacsim.util.debug_draw._debug_draw as isaac_debug_draw
-
-                    self.debug_draw = isaac_debug_draw.acquire_debug_draw_interface()
-                except ImportError:
-                    omni.log.info("isaacsim.util.debug_draw module not found. Debug visualization will be limited.")
+                LINE_MARKER_CFG = VisualizationMarkersCfg(
+                    prim_path="/World/Visuals/frame_transformer_lines",
```

Comment:
Is it possible to have these lines as additional marker type for `visualizer_cfg`? Then you can set them together with the frames by passing `marker_indices` as well to `VisualizationMarker.visualize()` operation. This also keeps it cleaner when toggling the visibility of these markers.

Reply:
@Mayankm96 Thanks, that’s a good point. I’ve reflected a change. Let me know if you have further suggestions.